test/unit/puppetlabs/services/master/master_core_test.clj

```diff
@@ -189,6 +190,31 @@
                         "supported"]
                        "bogus"}]})))))))
 
+(deftest valid-static-file-path-test
+  (let [valid-paths ["modules/foo/files/bar.txt"
+                     "modules/foo/files/bar"
+                     "modules/foo/files/bar/baz.txt"
+                     "modules/foo/files/bar/more/path/elements/baz.txt"]
+        invalid-paths ["modules/foo/manifests/bar.pp"
+                       "manifests/site.pp"
+                       "environments/foo/bar/files"
+                       "environments/../manifests/files/site.pp"
+                       "environments/../modules/foo/lib/puppet/parser/functions/site.rb"
+                       "environments/production/files/~/.bash_profile"
+                       "environments/../modules/foo/files/site.pp"
+                       "environments/production/modules/foo/files/../../../../../../site.pp"
+                       "environments/production/modules/foo/files/bar.txt"
+                       "environments/production/modules/foo/files/..conf..d../..bar..txt.."
+                       "environments/test/modules/foo/files/bar/baz.txt"
+                       "environments/dev/modules/foo/files/path/to/file/something.txt"]
```

Comment:
Might want to reject `modules/%2E%2E/manifests/site.pp` to make sure there's no confusion about URL decoding. Also `modules/\u002e\u002e/manifests/site.pp`. Or funny business like:

```
irb(main):001:0> puts "h\u0008.i\u0008."
..
```